### PR TITLE
Refactored the code to use Array.prototype.reduce

### PR DESCRIPTION
**Pull request by n4sa**
I figured you wanted to simplify this a bit. I noticed the extensive use of for-loops, thought I'd offer a different solution. It seems to pass all of your tests as well.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,50 +1,25 @@
...
-    , el
-
-  for(; i < l; ++i) {
-    keys = keys.concat(Object.keys(objs[i]))
-  }
-
-  for(i = 0, l = keys.length; i < l; ++i) {
-    if(keys.lastIndexOf(keys[i]) !== i) {
-      keys.splice(i, -1)
+  var objs = [].slice.call(arguments),
```
Comment (jarofghosts) [inline index.js]: can you fix this to use comma-first?

File: index.js [hunk 2]
```diff
@@ -1,50 +1,25 @@
...
+      len = objs.length
+
+  if(!len) return {}
+  if(len === 1) return objs.shift()
+
+  function mergeObjects(curr, next) {
+    var keys = Object.keys(curr),
+        key
```
Comment (jarofghosts) [inline index.js]: comma-first here as well plz

File: index.js [hunk 3]
```diff
@@ -1,50 +1,25 @@
...
+
+    for(var i = 0, len = keys.length; i < len; ++i) {
+      key = keys[i]
+      if(next.hasOwnProperty(key))
```
Comment (jarofghosts) [inline index.js]: i prefer braces for my conditionals if they are multi-line

File: index.js [hunk 4]
```diff
@@ -1,50 +1,25 @@
...
-        convert = true
-      }
-    }
-
-    if(keyResult.length === 1 && !convert) keyResult = keyResult[0]
-    result[attr] = keyResult
-  }
-
-  return result
+  return objs.reduce(mergeObjects)
```
Comment (jarofghosts) [inline index.js]: would rather this statement went above the function declaration for `mergeObjects`

**Comment by jarofghosts** [comment]
some style nits, but looks good otherwise

**Comment by n4sa** [comment]
There we go, I think I fixed everything correctly. I probably should have tested if this solution was more optimal than your previous one before making the pull request, it seems that's what you were going for. 

I just did a few performance.now() tests in Chrome after fixing the style changes and mine seemed to perform a little bit faster.

**Pull request merged**